tracks/python/exercises/saddle-points/mentoring.md

```diff
@@ -0,0 +1,195 @@
+suggest that they experiment with different checks to see what works.
+
+```
+# Suggested solution
+if max(row) == min(row[column_num] for row in matrix):
+
+# Alternative solution:
+def get_column(matrix, col_index):
+  return row[col_index] for row in matrix
+if max(row) == min(get_column(matrix, col_index):
+```
+
+### Returning the result
+
+The task requires returning a `list` of `dict` objects, one for each 
+saddle point, with the keys "row" and "column" and the values being
+the one-based index (i.e. the location, not the value, of the saddle 
+points).
+
+```
+result.append({"row": row_num + 1, "column": column_num + 1})
+    return result
+```
+
+There are various ways to structure the code for this problem; 
+do not insist that they use a particular approach if their
+chosen approach as also valid. Shorter code is not the only 
+goal here - readability is a key challenge in this task, so 
+don't make suggestions that would shorten the code at the 
+expense of readability. That said, many students will end up 
+writing much more code than is required - try to judge when 
+the extra length is just stepping through the problem and 
+where it could be best improved. Asking them to break of 
+complicated bit of code into separate functions can help 
+them work through each part without being overwhelmed - they 
+can recombine the code afterwards if desired.
+
+Students often create move variables that they need when
+looping, particularly when using multiple loops. They may
+also use unhelpful names for their variables, possibly 
+because they have too many to name. Encourage them to 
+give their variables descriptive names, and share common
+naming conventions such as the use of `i` and `j` for 
+indicies. 
+
+If their solution is difficult for you to understand due
+to the use of many confusing or single letter variables, 
+don't hesitate to start with this advice and wait for them 
+to fix this before giving further suggestions - there's no 
+point trying to parse their ninja code if you're going 
+to ask them to fix the issue anyway, and often making the 
+variable names clearer will help them improve the solution 
+on their own.
+
+Students will often include additional checks, such as 
+checking that the values are numeric or that the given
+type is a list. The only checks that are required to pass
+the tests are that the columns are all the same length,
+and perhaps to test for an empty matrix depending on the 
+approach. Hoever, there is nothing wrong with including 
+additional checks and you shouldn't ask the student to 
+remove them unless they are truly redundent (i.e. dead 
+code) or cause valid matricies to be rejected.
+
+## Talking points
+
+This problem potentially introduces several new concepts to
+the student, including:
+- inner loops
+- list comprehensions
+- the built-in functions `all`, `max`, `min` and `enumerate`
+
+Don't hesitate to introduce these concepts, as none of 
```

Comment:
These two paragraphs are also more of a "how to mentor" than "notes for this exercise".